src/sentry/usage_accountant/accountant.py

```diff
@@ -0,0 +1,72 @@
+"""
+This module is meant to manage the lifecycle of the UsageAccumulator
+to record shared resource usage amount.
+
+Since the UsageAccumulator depends on an Arroyo Kafka producer, the
+lifecycle is critical as the Threadpool has to be closed upon exit
+and the producer needs to be flushed to avoid loosing data.
+"""
+
+import atexit
+from typing import Optional
+
+from arroyo.backends.abstract import Producer
+from arroyo.backends.kafka import KafkaPayload, KafkaProducer, build_kafka_configuration
+from django.conf import settings
+from usageaccountant import UsageAccumulator, UsageUnit
+
+from sentry.options import get
+from sentry.utils.kafka_config import get_kafka_producer_cluster_options, get_topic_definition
+
+_accountant_backend: Optional[UsageAccumulator] = None
+
+
+def init_backend(producer: Producer[KafkaPayload]) -> UsageAccumulator:
+    """
+    This method should be used externally only in tests to fit a
+    mock producer.
+    """
+    global _accountant_backend
+
+    assert _accountant_backend is None, "Accountant already initialized once."
+    _accountant_backend = UsageAccumulator(producer=producer)
+    atexit.register(_shutdown)
+
+
+def _shutdown() -> None:
+    global _accountant_backend
+    _accountant_backend.flush()
+    _accountant_backend.close()
+
+
+def record(
```

Comment:
@untitaker, @lynnagara  I was in doubt what the right way to manage the life cycle would be. This library relies on a consumer so somebody needs to flush and close it before exisitng. Flushing is to avoid data loss, closing is required otherwise the process hangs.

This should be used in multiple places: celery workers, consumers, apis etc. 
There is not strict requirement to have one abstraction for all of them, but I noticed how more complex managing the lifecycle in an arroyo strategy is than using this simple `atexit` and destructor approach. 
I think I saw two possible ways:
- The one you see in code
- The arroyo one where the UsageAccountant lifecycle is tied to the life of the strategy.

In this scenario the simple atexit approach seems the best one: it works in multi process environments without touching the initializer and I do not have to add the lifecycle management in the strategy factory.

I wonder whether this should be the default way to manage resources consumers need to use and that do not need to be rebuilt upon rebalancing.

Reply:
> I noticed how more complex managing the lifecycle in an arroyo strategy is than using this simple atexit and destructor approach.

you have no control over shutdown timeouts with the global singleton approach, or the order in which things are being shutdown. to me atexit is probably the way to go if you need it in multiple backends, but it's a lot more indirect too, and testability becomes harder

also keep in mind that we don't have a comprehensive overview over where atexit _actually works_ -- as said in previous threads it's possible it does not work in celery at all, we just had no reason to check so far

Reply:
I'll add some logs to the shutdown process to see whether this is actually called.